internal/bminventory/inventory_test.go

```diff
@@ -10018,6 +10018,110 @@ var _ = Describe("TestRegisterCluster", func() {
 		ctrl.Finish()
 	})
 
+	// Testing the default network type setting during register flow, with various combinations of platforms,
+	// OpenShift version and high availability modes
+	// Note: SNO is supported starting OpenShift v4.8, and currently only supports Platform "none"
+	It("Default Network Type Setting", func() {
+		type clusterParamsForTest struct {
+			OpenShiftVersion     *string
+			HighAvailabilityMode *string
+			DesiredNetworkType   string
+			Platform             models.Platform
```

Comment:
I don't see how platform is involved in `getDefaultNetworkType`, why is it being varied here in the test vectors?